src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIRealtimeExtensions.cs

```diff
@@ -0,0 +1,146 @@
+using OpenAI.RealtimeConversation;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Provides extension methods for working with <see cref="RealtimeConversationSession"/> and related types.
+/// </summary>
+public static class OpenAIRealtimeExtensions
+{
+    /// <summary>
+    /// Converts a <see cref="AIFunction"/> into a <see cref="ConversationFunctionTool"/> so that
+    /// it can be used with <see cref="RealtimeConversationClient"/>.
+    /// </summary>
+    /// <returns>A <see cref="ConversationFunctionTool"/> that can be used with <see cref="RealtimeConversationClient"/>.</returns>
+    public static ConversationFunctionTool ToConversationFunctionTool(this AIFunction aiFunction)
+    {
+        _ = Throw.IfNull(aiFunction);
+
+        var parametersSchema = new ConversationFunctionToolParametersSchema
+        {
+            Type = "object",
+            Properties = aiFunction.Metadata.Parameters
+                .Where(p => p.Schema is JsonElement)
+                .ToDictionary(p => p.Name, p => (JsonElement)p.Schema!),
+            Required = aiFunction.Metadata.Parameters
+                .Where(p => p.IsRequired)
+                .Select(p => p.Name),
+        };
+
+        return new ConversationFunctionTool
+        {
+            Name = aiFunction.Metadata.Name,
+            Description = aiFunction.Metadata.Description,
+            Parameters = new BinaryData(JsonSerializer.SerializeToUtf8Bytes(
+                parametersSchema, OpenAIJsonContext.Default.ConversationFunctionToolParametersSchema))
+        };
+    }
+
+    /// <summary>
+    /// Handles tool calls.
+    ///
+    /// If the <paramref name="update"/> represents a tool call, calls the corresponding tool and
+    /// adds the result to the <paramref name="session"/>.
+    ///
+    /// If the <paramref name="update"/> represents the end of a response, checks if this was due
+    /// to a tool call and if so, instructs the <paramref name="session"/> to begin responding to it.
+    /// </summary>
+    /// <param name="session">The <see cref="RealtimeConversationSession"/>.</param>
+    /// <param name="update">The <see cref="ConversationUpdate"/> being processed.</param>
+    /// <param name="tools">The available tools.</param>
+    /// <param name="detailedErrors">An optional flag specifying whether to disclose detailed exception information to the model. The default value is <see langword="false"/>.</param>
+    /// <param name="jsonSerializerOptions">An optional <see cref="JsonSerializerOptions"/> that controls JSON handling.</param>
+    /// <param name="cancellationToken">An optional <see cref="CancellationToken"/>.</param>
+    /// <returns>A <see cref="Task"/> that represents the completion of processing, including invoking any asynchronous tools.</returns>
+    public static async Task HandleToolCallsAsync(
+        this RealtimeConversationSession session,
+        ConversationUpdate update,
+        IReadOnlyList<AIFunction> tools,
```

Comment:
Should we perhaps wrap the function list in a dedicated option type, in case more configuration is going to be necessary in the future?

Reply:
It's an interesting idea, but I'm trying to minimize the invention of new APIs/abstractions here since we don't yet know if/when we'd really create abstractions around realtime, and if we did, they would end up in a different package entirely.

We're definitely not trying to create some final APIs for realtime here - anything we deliver at this point is going to have breaking changes later when OpenAI's realtime APIs have further breaking changes. So I'd rather keep it simple and minimal and be ready to adapt as needed in the future.